src/ngx_http_lua_kong_module.c

```diff
@@ -624,3 +713,562 @@ ngx_http_lua_kong_get_upstream_ssl_verify(ngx_http_request_t *r,
 }
 
 #endif
+
+
+extern ngx_module_t ngx_http_proxy_module;
+
+
+static ngx_int_t
+ngx_http_proxy_body_output_filter(void *data, ngx_chain_t *in)
```

Comment:
for readability i'd suggest move this function and its companion structs that we copy paste from nginx to a separate file, and also add a macro check on the nginx core version we based off. since this function may change in the nginx core and the created request may out of sync

Reply:
this will also make it possible to support different versions of kong if we want to do that